core/auth/src/main/java/software/amazon/awssdk/auth/credentials/WebIdentityTokenCredentialsProvider.java

```diff
@@ -0,0 +1,173 @@
+import software.amazon.awssdk.auth.credentials.internal.WebIdentityTokenCredentialProperties;
+import software.amazon.awssdk.core.SdkSystemSetting;
+import software.amazon.awssdk.utils.ToString;
+
+/**
+ * A credential provider that will read web identity token file path, aws role arn
+ * and aws session name from system properties or environment variables for using
+ * web identity token credentials with STS. Use of this credentials provider requires
+ * the 'sts' module to be on the classpath.
+ */
+@SdkPublicApi
+public class WebIdentityTokenCredentialsProvider implements AwsCredentialsProvider {
+
+    private final AwsCredentialsProvider credentialsProvider;
+    private final RuntimeException loadException;
+
+    private WebIdentityTokenCredentialsProvider(BuilderImpl builder) {
+        AwsCredentialsProvider credentialsProvider = null;
+        RuntimeException loadException = null;
+
+        try {
+            String webIdentityTokenFilePath;
+            String roleArn;
+            Optional<String> roleSessionName;
+
+            // Don't allow a mix of environment variables and builder configured values
+            if (builder.webIdentityTokenFilePath != null) {
+                webIdentityTokenFilePath = builder.webIdentityTokenFilePath;
+                roleArn = builder.roleArn;
+                roleSessionName = Optional.ofNullable(builder.roleSessionName);
+            } else {
+                webIdentityTokenFilePath = trim(SdkSystemSetting.AWS_WEB_IDENTITY_TOKEN_FILE.getStringValueOrThrow());
+                roleArn = trim(SdkSystemSetting.AWS_ROLE_ARN.getStringValueOrThrow());
+                roleSessionName = SdkSystemSetting.AWS_ROLE_SESSION_NAME.getStringValue();
+            }
+
+            WebIdentityTokenCredentialProperties credentialProperties =
+                WebIdentityTokenCredentialProperties.builder()
+                                                    .roleArn(roleArn)
+                                                    .roleSessionName(roleSessionName.orElse(null))
+                                                    .webIdentityTokenFilePath(webIdentityTokenFilePath)
+                                                    .build();
+
+            credentialsProvider = WebIdentityCredentialsUtils.factory().create(credentialProperties);
+        } catch (RuntimeException e) {
+            // If we couldn't load the credentials provider for some reason, save an exception describing why. This exception
+            // will only be raised on calls to getCredentials. We don't want to raise an exception here because it may be
+            // expected (eg. in the default credential chain).
+            loadException = e;
+        }
+
+        if (loadException != null) {
+            this.loadException = loadException;
+            this.credentialsProvider = null;
+        } else {
+            this.credentialsProvider = credentialsProvider;
+            this.loadException = null;
+        }
```

Comment:
Both branches of this if statement will end up doing the same thing, so can we just make it...
```java
this.loadException = loadException;
this.credentialsProvider = credentialsProvider;
```
...?

Reply:
Yep yep